### PR TITLE
Feature-8470: Make attendee form ticket specific

### PR DESCRIPTION
**Pull request by lthanhhieu**
Fixes https://github.com/fossasia/open-event-frontend/issues/8740

#### Short description of what this resolves:
- Make attendee form ticket specific

#### Changes proposed in this pull request:
- Make attendee form ticket specific

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Review by cweitat — CHANGES_REQUESTED** [review]
fix issues

File: migrations/versions/rev-2023-06-15-08:53:12-d3c32d203ef1_.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Create Date: 2023-06-15 08:53:12.741108
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'd3c32d203ef1'
+down_revision = 'a45167cdc102'
```
Comment (norbusan) [inline migrations/versions/rev-2023-06-15-08:53:12-d3c32d203ef1_.py:13]: I am not sure this is the correct down revision, since there is
```
rev-2022-04-04-17:35:59-a45167cdc102_.py
rev-2023-06-12-13:39:49-410e54b84481_.py
```
so I guess it should be `410e54b84481`

File: app/api/helpers/custom_forms.py [hunk 2]
```diff
@@ -33,11 +34,23 @@ def get_schema(form_fields):
...
+            event_id=obj.event_id,
+            is_included=True,
+            form_id=ticket.form_id
+        ).all()
+    else:
+        form_fields = CustomForms.query.filter_by(
+            form=form,
+            event_id=obj.event_id,
+            is_included=True
+        ).all()
```
Comment (norbusan) [inline app/api/helpers/custom_forms.py:53]: Maybe something in the line of the following
(most probably not correct Python code, though!)
We don't want to duplicate arguments over and over
```
filter_args = ( form=form, event_id=obj.event_id, is_included=True )
if hasattr(obj, 'ticket_id'):
  filter_args += ( form_id=ticket.form_id )
form_fields = CustomForms.query.filter_by( *filter_args ).all()
```

File: app/api/routes.py [hunk 3]
```diff
@@ -688,7 +688,7 @@
     TicketDetail,
     'ticket_detail',
     '/tickets/<int:id>',
-    '/attendees/<int:attendee_id>/ticket',
+    '/attendees/<int:attendee_id>/ticket'
```
Comment (norbusan) [inline app/api/routes.py:691]: Please don't include unrelated changes in a PR. 
Final `,` are a good thing, since it keeps diffs small.
Please adjust your `black` configuration (or whatever formatter you are using)

File: app/api/routes.py [hunk 4]
```diff
@@ -796,7 +796,7 @@
     '/video-streams/<int:video_stream_id>/event',
     '/users-events-roles/<int:users_events_roles_id>/event',
     '/exhibitors/<int:exhibitor_id>/event',
-    '/speaker-invites/<int:speaker_invite_id>/event',
+    '/speaker-invites/<int:speaker_invite_id>/event'
```
Comment (norbusan) [inline app/api/routes.py:799]: Same as above

File: app/api/schema/custom_forms.py [hunk 5]
```diff
@@ -52,7 +52,7 @@ class Meta:
     is_included = fields.Boolean(default=False)
     is_public = fields.Boolean(default=False)
     position = fields.Integer(allow_none=True, default=0)
-    is_complex = fields.Boolean(dump_only=True)
+    is_complex = fields.Boolean(default=False)
```
Comment (norbusan) [inline app/api/schema/custom_forms.py:55]: That is a surprising change. The `dump_only` looks completely out of place, and `default=` looks correct.
How did that come in there?

File: migrations/versions/rev-2023-06-15-08:53:12-d3c32d203ef1_.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+Create Date: 2023-06-15 08:53:12.741108
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'd3c32d203ef1'
+down_revision = '410e54b84481'
```
Comment (norbusan) [inline migrations/versions/rev-2023-06-15-08:53:12-d3c32d203ef1_.py:13]: That is incorrect I think, we already merged another revision ... maybe ...

File: app/api/helpers/custom_forms.py (hunk 2 above)
Comment (lthanhhieu) [reply app/api/helpers/custom_forms.py]: Thank you for your suggestion, I have changed it as you want. That's good for us to clean code as well ;)

File: migrations/versions/rev-2023-06-15-08:53:12-d3c32d203ef1_.py (hunk 1 above)
Comment (lthanhhieu) [reply migrations/versions/rev-2023-06-15-08:53:12-d3c32d203ef1_.py]: It's has been resolved.

File: app/api/schema/custom_forms.py (hunk 5 above)
Comment (lthanhhieu) [reply app/api/schema/custom_forms.py:55]: After checked, I think that should be contained both. dump_only and default. I think that should not be modified after created; hence, dump_only still correct. I will add it back.

**Review by norbusan — APPROVED** [review]
LGTM

**Review by cweitat — APPROVED** [review]
